inc/dbconnection.class.php

```diff
@@ -53,22 +53,24 @@ static function getTypeName($nb = 0) {
     *
     * @since 9.1
     *
-    * @param string $host      The DB host
-    * @param string $user      The DB user
-    * @param string $password  The DB password
-    * @param string $DBname    The name of the DB
+    * @param string  $host                      The DB host
+    * @param string  $user                      The DB user
+    * @param string  $password                  The DB password
+    * @param string  $DBname                    The name of the DB
+    * @param boolean $use_utf8mb4               Flag that indicates if utf8mb4 charset/collation should be used
+    * @param boolean $log_deprecation_warnings  Flag that indicates if DB deprecation warnings should be logged
     *
     * @return boolean
-    *
-   **/
-   static function createMainConfig($host, $user, $password, $DBname, $use_utf8mb4 = false) {
+    */
+   static function createMainConfig($host, $user, $password, $DBname, $use_utf8mb4 = false, $log_deprecation_warnings = false) {
 
       $DB_str = "<?php\nclass DB extends DBmysql {\n" .
                 "   public \$dbhost      = '$host';\n" .
                 "   public \$dbuser      = '$user';\n" .
                 "   public \$dbpassword  = '". rawurlencode($password) . "';\n" .
                 "   public \$dbdefault   = '$DBname';\n" .
                 "   public \$use_utf8mb4 = " . ($use_utf8mb4 ? 'true' : 'false') . ";\n" .
+                "   public \$log_deprecation_warnings = " . ($log_deprecation_warnings ? 'true' : 'false') . ";\n" .
```

Comment:
As the property is present in the parent, could we write the line only when `$log_deprecation_warnings` is present.
I'm not sure the parameter will be usefull in production instances (Is the purpose of this is only catching with CI on our side ?) and set it with false default value will only bring questions on support.

Reply:
Indeed, no need to get support due to this.

While changing it, I take time to add unit tests on this, so I had to made some code refactoring.